apps/web/src/components/Composer/OpenActionsPreviews.tsx

```diff
@@ -0,0 +1,179 @@
+  ): Promise<any | undefined> => {
+    const nftOpenActionKit = getNftOpenActionKit();
+    const publicationContentUrls = getURLs(publicationContent);
+
+    try {
+      const calldata = await nftOpenActionKit.detectAndReturnCalldata({
+        contentURI: publicationContentUrls[0],
+        publishingClientProfileId: HEY_REFERRAL_PROFILE_ID
+      });
+
+      if (calldata) {
+        return {
+          unknownOpenAction: {
+            address: VerifiedOpenActionModules.DecentNFT,
+            data: calldata
+          }
+        };
+      } else {
+        return undefined;
+      }
+    } catch (error) {
+      console.error('Error fetching open action embed:', error);
+      return undefined;
+    }
+  };
+
+  const { data: nftOpenActionEmbed, isLoading: nftOpenActionEmbedLoading } =
+    useQuery({
+      enabled: Boolean(publicationContent),
+      queryFn: () => fetchnftOpenActionEmbed(publicationContent),
+      queryKey: ['fetchnftOpenActionEmbed', publicationContent]
+    });
+
+  const { data } = useQuery({
+    enabled: Boolean(url),
+    queryFn: async () => {
+      const response = await axios.get(`${HEY_API_URL}/oembed`, {
+        params: { url }
+      });
+      return response.data.oembed;
+    },
+    queryKey: ['getOembed', url],
+    refetchOnMount: false
+  });
+
+  useEffect(() => {
+    if (urls.length) {
+      removeAttribute(KNOWN_ATTRIBUTES.HIDE_OEMBED);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [urls.length]);
+
+  useEffect(() => {
+    if (nftOpenActionEmbed) {
+      setNftOpenActionEmbed(nftOpenActionEmbed);
+    } else {
+      setNftOpenActionEmbed(undefined);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [nftOpenActionEmbed]);
+
+  const og: OG = {
+    description: data?.description,
+    favicon: data?.url ? getFavicon(data.url) : '',
+    html: data?.html,
+    image: data?.image,
+    nft: data?.nft,
+    site: data?.site,
+    title: data?.title,
+    url: url as string
+  };
+
+  if (
+    !nftOpenActionEmbed ||
+    !urls.length ||
+    attachments.length ||
+    quotedPublication ||
+    getAttribute(KNOWN_ATTRIBUTES.HIDE_OEMBED)?.value === 'true'
+  ) {
+    return null;
+  }
+
+  if (nftOpenActionEmbed) {
+    return (
+      <div className="relative m-5">
+        <DecentOpenActionPreview
+          og={og}
+          openActionEmbedLoading={nftOpenActionEmbedLoading}
+        />
+        <div className="absolute top-0 m-3">
+          <button
+            className="rounded-full bg-gray-900 p-1.5 opacity-75"
+            onClick={() =>
+              addAttribute({
+                key: KNOWN_ATTRIBUTES.HIDE_OEMBED,
+                type: MetadataAttributeType.BOOLEAN,
+                value: 'true'
+              })
+            }
+            type="button"
+          >
+            <XMarkIcon className="size-4 text-white" />
+          </button>
+        </div>
+      </div>
+    );
+  }
+
+  if (!hasSwapOpenAction) {
```

Comment:
lets make a true check here, and default to return null if all false so it matches thee above pattern for nft check

true

true

false at last